x/crosschain/keeper/msg_add_to_outtx_tracker.go

```diff
@@ -0,0 +1,106 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	eth "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// AddToOutTxTracker adds a new record to the outbound transaction tracker.
+// only the admin policy account and the observer validators are authorized to broadcast this message.
+func (k msgServer) AddToOutTxTracker(goCtx context.Context, msg *types.MsgAddToOutTxTracker) (*types.MsgAddToOutTxTrackerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	chain := k.zetaObserverKeeper.GetParams(ctx).GetChainFromChainID(msg.ChainId)
+	if chain == nil {
+		return nil, observertypes.ErrSupportedChains
+	}
+	adminPolicyAccount := k.zetaObserverKeeper.GetParams(ctx).GetAdminPolicyAccount(observertypes.Policy_Type_group1)
+	isAdmin := msg.Creator == adminPolicyAccount
+
+	isObserver, err := k.zetaObserverKeeper.IsAuthorized(ctx, msg.Creator, chain)
+	if err != nil {
+		ctx.Logger().Error("Error while checking if the account is an observer", err)
+	}
+
+	isProven := false
+	if !(isAdmin || isObserver) && msg.Proof != nil {
+		txx, err := k.VerifyProof(ctx, msg.Proof, msg.BlockHash, msg.TxIndex, msg.ChainId)
+		if err != nil {
+			return nil, types.ErrCannotVerifyProof.Wrapf(err.Error())
+		}
+		err = k.VerifyOutTxTrackerProof(ctx, txx, msg.Nonce)
```

Comment:
I created two functions `ValidateEVMOutTxBody` and `ValidateBTCOutTxBody` to verify the `msg` `tss` against the original transaction body to prevent data mismatch in [Bitcoin header N proof PR](https://github.com/zeta-chain/node/pull/1263/files) in the file x/crosschain/keeper/keeper_out_tx_tracker.go. We'll have to unify the workflow a bit for both chains. Just a FYI, not a blocker.